process_test.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/getsentry/sentry-go"
+)
+
+func TestProcessLine(t *testing.T) {
+	_verbose = true
+	err := sentry.Init(sentry.ClientOptions{Debug: true})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	g := initGrokProcessor()
+
+	processLine(
+		`127.0.0.1 - - [23/Apr/2014:22:58:32 +0200] "GET /index.php HTTP/1.1" 404 207`,
+		[]string{"%{COMMONAPACHELOG}"},
+		g,
+		sentry.CurrentHub(),
+	)
+
+	processLine(
+		"",
+		[]string{"%{COMMONAPACHELOG}"},
+		g,
+		sentry.CurrentHub(),
+	)
```

Comment:
At the moment these calls to `processLine` don't really assert anything (other than not crashing). One potential solution here is to have something similar to [`TransportMock`](https://github.com/getsentry/sentry-go/blob/eb22fb532ae6b049ad2fc36bc10180b834ae38de/mocks_test.go#L24C1-L44), and then assert that the proper events were built via `Events()` method.
Lmk if you want to pursue that, otherwise this is already an improvement.

Reply:
I went with `TransportMock` to assert stuff.